lib/async/websocket/client.rb

```diff
@@ -90,14 +90,14 @@ def close
 				end
 			end
 			
-			def connect(authority, path, headers: nil, handler: Connection, extensions: ::Protocol::WebSocket::Extensions::Client.default, **options, &block)
+			def connect(authority, path, scheme: @delegate.scheme, headers: nil, handler: Connection, extensions: ::Protocol::WebSocket::Extensions::Client.default, **options, &block)
```

Comment:
This looks okay to me, but realistically speaking, is there ever going to be a case where we want to override the scheme in this way? We can always introduce an argument later, but maybe we can use `@delegate.scheme` on line 100 for now? WDYT?

Reply:
I don't presently have a concrete use case for the extra arg. It does pass through from `def self.connect(..., **options)` on line 55, so it's a little more versatile than at first glance. 

Because of that pass through, my intention was to parallel the usage of `Async::HTTP::Client` where high-level method calls already support overriding `scheme:` apart from the endpoint. If that parallelism seems worth it, then perhaps keep it? Otherwise, it's no problem to remove the arg. I'll let you make the call. 😄

Reply:
```suggestion
			def connect(authority, path, headers: nil, handler: Connection, extensions: ::Protocol::WebSocket::Extensions::Client.default, **options, &block)
				scheme = @delegate.scheme
```